pkg/ddc/thin/referencedataset/engine.go

```diff
@@ -186,55 +192,64 @@ func (e *ReferenceDatasetEngine) Setup(ctx cruntime.ReconcileRequestContext) (re
 
 // Shutdown and clean up the engine
 func (e *ReferenceDatasetEngine) Shutdown() (err error) {
-	// 1. delete this dataset to mounted dataset DatasetRef field
+	// 1. delete this dataset in physical dataset DatasetRef field
 	datasetRefName := base.GetDatasetRefName(e.name, e.namespace)
 
-	mountedRuntimeInfo, err := e.getMountedRuntimeInfo()
-	if err != nil {
-		return err
-	}
-	mountedDataset, err := utils.GetDataset(e.Client, mountedRuntimeInfo.GetName(), mountedRuntimeInfo.GetNamespace())
-	if err != nil {
+	physicalRuntimeInfo, err := e.getPhysicalRuntimeInfo()
+	if err != nil && utils.IgnoreNotFound(err) != nil {
 		return err
 	}
 
-	if utils.ContainsString(mountedDataset.Status.DatasetRef, datasetRefName) {
-		newDataset := mountedDataset.DeepCopy()
-		newDataset.Status.DatasetRef = utils.RemoveString(newDataset.Status.DatasetRef, datasetRefName)
-		err := e.Client.Status().Update(context.TODO(), newDataset)
+	if physicalRuntimeInfo != nil {
```

Comment:
Please add log message to indicate `physicalRuntimeInfo == nil` case so that we can know corner case happened.